pkg/gofr/cmd/request.go

```diff
@@ -101,23 +101,26 @@ func (r *Request) Bind(i interface{}) error {
 	ps := reflect.ValueOf(i)
 	// struct
 	s := ps.Elem()
-	if s.Kind() == reflect.Struct {
-		for k, v := range r.params {
-			f := s.FieldByName(k)
-			// A Value can be changed only if it is addressable and not unexported struct field
-			if f.IsValid() && f.CanSet() {
-				//nolint:exhaustive // no need to add other cases
-				switch f.Kind() {
-				case reflect.String:
-					f.SetString(v)
-				case reflect.Bool:
-					if v == trueString {
-						f.SetBool(true)
-					}
-				case reflect.Int:
-					n, _ := strconv.Atoi(v)
-					f.SetInt(int64(n))
+
+	if s.Kind() != reflect.Struct {
+		return nil
+	}
+
+	for k, v := range r.params {
+		f := s.FieldByName(k)
+		// A Value can be changed only if it is addressable and not unexported struct field
+		if f.IsValid() && f.CanSet() {
+			//nolint:exhaustive // no need to add other cases
```

Comment:
To reduce the complexity you can rewrite it like that


```suggestion
		if !f.IsValid() || !f.CanSet() {
			continue
		}
		//nolint:exhaustive // no need to add other cases
```

This way you start the switch at the same level as the if

Reply:
ah that makes sense, thanks!